extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/DevServicesElasticsearchProcessor.java

```diff
@@ -0,0 +1,180 @@
+
+        if (devService != null) {
+            boolean shouldShutdownTheBroker = !configuration.equals(cfg);
+            if (!shouldShutdownTheBroker) {
+                return devService.toBuildItem();
+            }
+            shutdownElasticsearch();
+            cfg = null;
+        }
+
+        StartupLogCompressor compressor = new StartupLogCompressor(
+                (launchMode.isTest() ? "(test) " : "") + "Elasticsearch Dev Services Starting:",
+                consoleInstalledBuildItem, loggingSetupBuildItem);
+        try {
+            devService = startElasticsearch(configuration, launchMode,
+                    !devServicesSharedNetworkBuildItem.isEmpty(),
+                    devServicesConfig.timeout);
+            compressor.close();
+        } catch (Throwable t) {
+            compressor.closeAndDumpCaptured();
+            throw new RuntimeException(t);
+        }
+
+        if (devService == null) {
+            return null;
+        }
+
+        // Configure the watch dog
+        if (first) {
+            first = false;
+            Runnable closeTask = () -> {
+                if (devService != null) {
+                    shutdownElasticsearch();
+                }
+                first = true;
+                devService = null;
+                cfg = null;
+            };
+            closeBuildItem.addCloseTask(closeTask, true);
+        }
+        cfg = configuration;
+
+        if (devService.isOwner()) {
+            log.infof(
+                    "Dev Services for Elasticsearch started. Other Quarkus applications in dev mode will find the "
+                            + "server automatically. For Quarkus applications in production mode, you can connect to"
+                            + " this by starting your application with -Dquarkus.elasticsearch.hosts=%s",
+                    getElasticsearchHosts());
+        }
+        return devService.toBuildItem();
+    }
+
+    public static String getElasticsearchHosts() {
+        return devService.getConfig().get(ELASTICSEARCH_HOSTS);
+    }
+
+    private void shutdownElasticsearch() {
+        if (devService != null) {
+            try {
+                devService.close();
+            } catch (Throwable e) {
+                log.error("Failed to stop the Elasticsearch server", e);
+            } finally {
+                devService = null;
+            }
+        }
+    }
+
+    private DevServicesResultBuildItem.RunningDevService startElasticsearch(ElasticsearchDevServicesBuildTimeConfig config,
+            LaunchModeBuildItem launchMode, boolean useSharedNetwork, Optional<Duration> timeout) {
+        if (!config.enabled.orElse(true)) {
+            // explicitly disabled
+            log.debug("Not starting dev services for Elasticsearch, as it has been disabled in the config.");
+            return null;
+        }
+
+        // Check if kafka.bootstrap.servers is set
```

Comment:
Outdated comment.

Reply:
Yes, you can clearly see from where I copy/paste the code ;)

Reply:
fixed